src/quick_sort.py

```diff
@@ -0,0 +1,68 @@
+"""Quick sort a list."""
+import time
+from random import randint
+
+
+def quick_sort(num_list):
+    """Use the quick sort method to sort a list of numbers."""
+    if len(num_list) > 1:
+
+        pivot = num_list[0]
+        left_mark = 1
+        right_mark = len(num_list) - 1
+
+        while left_mark <= right_mark:
+            if num_list[left_mark] > pivot and num_list[right_mark] < pivot:
+                num_list[left_mark], num_list[right_mark] =\
+                    num_list[right_mark], num_list[left_mark]
+                left_mark += 1
+                right_mark -= 1
+            elif num_list[left_mark] <= pivot and num_list[right_mark] < pivot:
+                left_mark += 1
+            elif num_list[left_mark] > pivot and num_list[right_mark] >= pivot:
+                right_mark -= 1
+            elif num_list[left_mark] <= pivot and num_list[right_mark] >= pivot:
+                left_mark += 1
+                right_mark -= 1
+
+        num_list[0], num_list[right_mark] = num_list[right_mark], num_list[0]
+
+        divider = right_mark + 1
+        left = quick_sort(num_list[:right_mark])
+        right = quick_sort(num_list[divider:])
+        return left + [num_list[right_mark]] + right
+    else:
+        return num_list
+
+
+if __name__ == '__main__':
+    print('\nCASE 1: A small list to be sorted:\n'
```

Comment:
You're supposed to have best case, worst case, and an average case in your timing.